### PR TITLE
ingress annotation changes specifially to support docker registry

### PR DESCRIPTION
**Pull request by zendril**
@eldada @jainishshah17 
**What this PR does / why we need it**:
The current annotation support in the ingress templates iterates over items, which breaks certain concepts like configuration-snippets.

This enables the annotation section to be passed in verbatim (or omitted altogether).

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #
No existing issue number, however it is tied to an internal jfrog support ticket.

**Special notes for your reviewer**:
By doing this, it introduces a warning from lint:
`2018/10/17 14:27:13 warning: destination for annotations is a table. Ignoring non-table value <nil>`

In my research (including talking to people on the #helm-users slack channel), this appears to be a known issue and a bug in Helm 2 which they tried to fix but rolled it back because it broke other dependencies like prometheus.

If you can find a way around this I would be happy to make the change and update the PR, but I'm currently not seeing one.

### CODE REVIEW

**Comment by zendril** [comment]
Done. 
You would think I'd remember to do it every time by now..  guess not :/

File: stable/artifactory-ha/README.md [hunk 1]
```diff
@@ -495,6 +496,31 @@ Include the secret's name, along with the desired hostnames, in the Artifactory
           - artifactory.domain.com
 ```
 
+### Ingress annotations
```
Comment (jainishshah17) [inline stable/artifactory-ha/README.md]: @zendril We should specify that these annotations are only to use Artifactory as Docker Registry with   Repository Path as a reverse proxy method.

**Comment by jainishshah17** [comment]
@zendril Also please fix conflicts.

File: stable/artifactory-ha/README.md (hunk 1 above)
Comment (zendril) [reply stable/artifactory-ha/README.md]: Fair point. I'll update.

Comment (zendril) [reply stable/artifactory-ha/README.md]: @jainishshah17 Done and pushed.

**Comment by zendril** [comment]
> @zendril Also please fix conflicts.

@jainishshah17 Done and pushed.

**Pull request merged**